core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala

```diff
@@ -230,12 +251,16 @@ class NewHadoopRDD[K, V](
     super.persist(storageLevel)
   }
 
-
-  def getConf: Configuration = confBroadcast.value.value
```

Comment:
Hmm.. ok. What do you think about keeping the public getConf as is and create a protected "getOptionallyClonedConf" that does cloning optionally based on the flag? However, the downside is that it's vulnerable to human mistakes in the future when someone accidentally uses getConf as opposed to getOptionallyClonedConf for computation. I'm slightly inclined to the getOptionallyClonedConf route for the possible backcompat break.